.script/tests/detectionTemplateStructureValidation/Models/ValidEntityMappingsAttribute.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Sentinel.Analytics.Management.AnalyticsManagement.Contracts.Model.ARM.ModelValidation
+{
+    public class ValidEntityMappingsAttribute : ValidationAttribute
+    {
+        private readonly int _entityMappingsMinLength;
+        private readonly int _entityMappingsMaxLength;
+        private readonly int _fieldMappingsMinLength;
+        private readonly int _fieldMappingsMaxLength;
+        private readonly int _laColumnNameMaxLength = 500; // 500 is the max length of a column name in LA
+
+        // field mapping column name should start with a letter or an underscore and contain only alphanumeric English characters (i.e. [a-zA-Z0-9_])
+        private readonly Regex _laColumnNameRegex = new Regex("^[a-zA-Z_]+\\w*$");
+
+        public ValidEntityMappingsAttribute(int entityMappingsMinLength, int entityMappingsMaxLength, int fieldMappingsMinLength, int fieldMappingsMaxLength)
+        {
+            _entityMappingsMinLength = entityMappingsMinLength;
+            _entityMappingsMaxLength = entityMappingsMaxLength;
+            _fieldMappingsMinLength = fieldMappingsMinLength;
+            _fieldMappingsMaxLength = fieldMappingsMaxLength;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if (value == null)
+            {
+                return ValidationResult.Success;
+            }
+
+            var entityMappings = (List<EntityMapping>)value;
```

Comment:
consider try catch for casting exception

Reply:
we won't reach here if the deserialization fails. So we are sure it casts.
Even if it doesn't we would throw anyway in the catch since we want to fail the validation in that case.. so it won't make any difference